.github/workflows/ci.yml

```diff
@@ -0,0 +1,130 @@
+name: Plio CI
+
+on:
+  pull_request:
+  push:
+    branches: [master]
+
+jobs:
+  pre-commit:
+    name: Pre-commit
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+      - uses: pre-commit/action@v2.0.0
+
+  integration-tests:
+    name: Integration tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Plio Frontend
+        run: |
+          # clone the project
+          mkdir -p projects/ && cd projects/
+          git clone https://github.com/avantifellows/plio-frontend/
+          cd plio-frontend/
+
+          # check branch and switch to branch if exists
+          if [ `git branch --list --remote origin/${{ github.head_ref }}` ]
+          then
+            echo "Switching to branch ${{ github.head_ref }}."
+            git checkout ${{ github.head_ref }}
+            git pull origin ${{ github.head_ref }}
+          else
+            echo "Branch not found. Going with default branch."
+          fi
```

Comment:
can we not save the output of branch name from this step and re-use in the steps below?

Reply:
@dalmia I think there is some confusion here.

So the branch name is easily available in GitHub action using `github.head_ref` option. Inside the if condition, I'm checking if the branch is available in the remote repository and if it does, it switches to that branch.

So we don't need the output to be re-used as each repo will have to check the branch existence in their own remote URLs. Let me know if this makes sense.